structuretoolkit/visualize.py

```diff
@@ -807,3 +809,78 @@ def _get_flattened_orientation(
     flattened_orientation[:3, :3] = _get_orientation(view_plane)
 
     return (distance_from_camera * flattened_orientation).ravel().tolist()
+
+
+def plot_isosurface(
+    mesh,
+    value,
+    cell: Optional[Union[Atoms, list, np.ndarray, float]] = None,
+    structure_plot: Optional["plotly.graph_objs._figure.Figure"] = None,
+    isomin: Optional[float] = None,
+    isomax: Optional[float] = None,
+    surface_fill: Optional[float] = None,
+    opacity: Optional[float] = None,
+    surface_count: Optional[int] = 5,
+    colorbar_nticks: Optional[int] = None,
+    caps: Optional[dict] = dict(x_show=False, y_show=False, z_show=False),
+    colorscale: Optional[str] = None,
+    height: Optional[float] = 600,
+    camera: Optional[str] = "orthographic",
+    **kwargs,
+):
+    """
+    Make a mesh plot
+
+    Args:
+        mesh (numpy.ndarray): Mesh grid. Must have a shape of (3, nx, ny, nz).
+            It can be generated from structuretoolkit.create_mesh
+        value: (numpy.ndarray): Value to plot. Must have a shape of (nx, ny, nz)
+        cell (Atoms|ndarray|list|float|tuple): Cell, ignored if
+            `structure_plot` is given
+        structure_plot (plotly.graph_objs._figure.Figure): Plot of the
+            structure to overlay. You should basically always use
+            structuretoolkit.plot3d(structure, mode="plotly")
+        isomin(float): Min color value
+        isomax(float): Max color value
+        surface_fill(float): Polygonal filling of the surface to choose between
+            0 and 1
+        opacity(float): Opacity
+        surface_count(int): Number of isosurfaces, 5 by default, which means
```

Comment:
I believe you...but why? I would have thought it was `2` just like it was until the last commit (at time of writing). This is so weird it might warrant some sort of "I know, I know, but seriously, it's because X" in the docstring.

Reply:
yeah I thought about it a little bit XD The problem is that somehow plotly sets the two layers for the highest value and the lowest value. It makes only sense if there are multiple points with the max/min value, but that's rarely the case. So even with 5, in reality you would see only 3 in the middle. I also thought about changing `isomax` and `isomin`, but that felt a bit like too much manipulation.

Reply:
That's very fair. Maybe something like `, which actually means only min and max because of details of how plotly handles the levels` to explain the weirdness just a little?